File: codegenerator/benchmarks/cpp/IncreasingSet.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BSet.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef IncreasingSet_H
#define IncreasingSet_H

using namespace std;

class IncreasingSet {



    private:



        BInteger counter;
        BSet<BInteger > set;

        bool initialized = false;

    public:

        IncreasingSet(){}

        void initialize() {
            if(initialized) {
                throw runtime_error("Machine is already initialized");
            }
            counter = static_cast<BInteger >((BInteger(0)));
            set = static_cast<BSet<BInteger > >((BSet<BInteger >((vector<BInteger>){(BInteger(1)), (BInteger(2)), (BInteger(3))})));
            initialized = true;
        }

        void simulate() {
            if(!initialized) {
                throw runtime_error("Machine was not initialized");
            }
            while((counter.less((BInteger(25000)))).booleanValue()) {
                set = static_cast<BSet<BInteger > >(set._union((BSet<BInteger >((vector<BInteger>){counter}))));
                counter = static_cast<BInteger >(counter.plus((BInteger(1))));
            }
        }

};

int main() {
    clock_t start,finish;
    double time;
    IncreasingSet exec;
    exec.initialize();
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

File: codegenerator/benchmarks/java/primitive_integer_wrapper/IncreasingSet.java
import de.hhu.stups.btypes.BSet;
import de.hhu.stups.btypes.BInteger;

public class IncreasingSet {




    private BInteger counter;
    private BSet set;

    private boolean initialized = false;

    public void initialize() {
        if(initialized) {
            throw new RuntimeException("Machine is already initialized");
        }
        counter = (BInteger) new BInteger(0);
        set = (BSet) new BSet(new BInteger(1), new BInteger(2), new BInteger(3));
        initialized = true;
    }

    public void simulate() {
        if(!initialized) {
            throw new RuntimeException("Machine was not initialized");
        }
        while((counter.less(new BInteger(25000))).booleanValue()) {
            set = (BSet) set.union(new BSet(counter));
            counter = (BInteger) counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        IncreasingSet exec = new IncreasingSet();
        exec.initialize();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}
